app/commands/solution/complete.rb

```diff
@@ -17,7 +17,7 @@ def call
       # we only go through the checks in the job if the exercise is hello-world.
       # But at the job level, not at this level. So passing some args etc?
       %i[anybody_there all_your_base whatever lackadaisical].each do |badge|
-        AwardBadgeJob.perform_later(user, badge)
+        AwardBadgeJob.perform_later(user, badge, exercise: exercise.slug)
```

Comment:
The magic here is the passing of the `exercise` keyword. 
All the badges that are to be awarded here are specific to an exercise, so with just the passing of the exercise slug they have enough information to short-circuit their behavior.